core/rawdb/freezer_table.go

```diff
@@ -354,17 +364,25 @@ func (t *freezerTable) truncate(items uint64) error {
 		log = t.logger.Warn // Only loud warn if we delete multiple items
 	}
 	log("Truncating freezer table", "items", existing, "limit", items)
-	if err := truncateFreezerFile(t.index, int64(items+1)*indexEntrySize); err != nil {
-		return err
-	}
+
+	// Truncate the index file first, the tail position is also considered
+	// when calculating the new freezer table length.
 	// Calculate the new expected size of the data file and truncate it
-	buffer := make([]byte, indexEntrySize)
-	if _, err := t.index.ReadAt(buffer, int64(items*indexEntrySize)); err != nil {
+	length := items - t.itemOffset.Load()
+	if err := truncateFreezerFile(t.index, int64(length+1)*indexEntrySize); err != nil {
```

Comment:
Before we only ReadAt index file, but right now in  trunecateFreezer, we need to resize the index file firstly(remove all entries before itemOffets, then seek to  then the length should be indexOffset-> items, and go to the end of file for append new

Reply:
@Francesco4203 the other reason, I though, because they introduce the truncateTail, so the index of tail should be referred  when calculating  truncate the current index file too,  before, we only truncateAncient( which work mostly for truncatehead)  so we need to keep the items of index (the truncate function doesn't change the offset of file)